### ISSUE TITLE
[BUG] .not.toBeVisible with small timeouts always fails regardless of page content

### ISSUE THREAD

**Issue by AlexDaniel** [microsoft/playwright#18859]
**Context:**
- Playwright Version: Version 1.25.0
- Operating System: Linux
- Node.js version: v18.10.0
- Browser: Chromium, but probably all

This test ALWAYS fails, regardless of page content:
```js
await expect(page.locator('text=this text does not exist, or it does, there is no difference, the test will fail')).not.toBeVisible({ timeout: 10 })
```

I think it's rather obvious that using `expect` should perform some work and not just be a no-op (even if the specified timeout is small).

Previous ticket (https://github.com/microsoft/playwright/issues/18815) was closed because there was some confusion about the intent of such code (and also because I wasn't very precise in my description). The intent of small timeouts is completely irrelevant, the retry logic should not suddenly break just because a smaller timeout is passed. Note that timeouts higher than 10ms seem to be flaky (instead of always failing like it is for 10ms), so I'm not sure what is the usable range of timeouts. 50ms seems to work correctly, while 20ms has about 50% chance of working.

**Comment by pavelfeldman** [comment]
I did misunderstand the original request, this one makes it clear, thanks!

**Comment by pavelfeldman** [comment]
I'm not sure the fix will be to your expectations. Since we must respect the given timeout, the fix will make it clear that Timeout 10ms was exceeded. We can't know if the element is hidden or visible w/o exceeding the given small timeout.

**Comment by AlexDaniel** [comment]
> I'm not sure the fix will be to your expectations. Since we must respect the given timeout, the fix will make it clear that Timeout 10ms was exceeded. We can't know if the element is hidden or visible w/o exceeding the given small timeout.

But isn't playwright supposed to perform at least one check, even if the timeout was exceeded?

**Comment by pavelfeldman** [comment]
Not really, the contract is to respect the timeout. If you don't care about the timeout or retry, you can just query the state.

**Comment by AlexDaniel** [comment]
@pavelfeldman that is the wrong contract. When I write a test, I expect it to do something, it shouldn't just fail randomly. If `expect` is allowed to do nothing at all and fail just because the machine is slow, then what is the acceptable range for timeouts and how would I as a user ever know it?

**Comment by pavelfeldman** [comment]
Unfortunately, going the other way would raise the opposite complaints, as in, I passed `timeout: 10`, but the actions stalled for a minute. We need to balance those somehow. It looks like you are misusing the API though, if you want to make a call, you should not misuse the timer and should rather just make that call.

**Comment by dgozman** [comment]
Fixed by #18895.
